### ISSUE TITLE
Blazor WASM Catch Unhandled Exception

### ISSUE THREAD

**Issue by TanayParikh** [dotnet/aspnetcore#42347]
Looks like this is an issue impacting the community: https://www.reddit.com/r/Blazor/comments/vhrrl1/blazor_wasm_how_to_break_on_unhandled_exception/

Ideally we can hook into the `System.AppDomain.CurrentDomain.UnhandledException` as we do with WebView for WinForms / WPF. 

Repro:

`Index.razor`:

```razor
<button @onclick="ThrowException">Throw Exception</button>

@code {
    public void ThrowException()
    {
        throw new Exception("throwing new ex");
    }
}
```

`Program.cs`:

```razor
System.AppDomain.CurrentDomain.UnhandledException += (sender, error) =>
{
    System.Console.WriteLine("!!! Encountered exception !!!");
};
```

Kind of related: https://github.com/dotnet/maui/issues/5535

**Comment by javiercn** [comment]
@TanayParikh if its about breaking when an exception happens, we shouldn't do anything in the framework for it, "Break for first chance exceptions" is a debugger feature. We should make sure that it works in all platforms.

**Comment by jasonswearingen** [comment]
Yes the issue described in that reddit post is the inability to break on unhandled exceptions.

**Comment by TanayParikh** [comment]
> if its about breaking when an exception happens, we shouldn't do anything in the framework for it, "Break for first chance exceptions" is a debugger feature. 

> Yes the issue described in that reddit post is the inability to break on unhandled exceptions.

Just making sure we aren't talking about two different things. I believe @javiercn is saying the framework doesn't have to automatically support breaking on unhandled exceptions using a first chance mechanism like we see with C# in VS. 

@jasonswearingen are you perhaps suggesting that it's not possible to manually break on unhandled exceptions? In that case, wouldn't a mechanism like `System.AppDomain.CurrentDomain.UnhandledException` be ideal so that users are able to set breakpoints for unhandled exceptions should they occur?

**Comment by jasonswearingen** [comment]
> @jasonswearingen are you perhaps suggesting that it's not possible to manually break on unhandled exceptions at all? In that case, wouldn't a mechanism like `System.AppDomain.CurrentDomain.UnhandledException` be ideal so that users are able to set breakpoints for unhandled exceptions should they occur?

Perhaps that is the best workaround for the issue, but I prefer not to speculate, as perhaps setting a breakpoint in a global exception handler might destroy state at the callsite that a proper "break on unhandled exception" would preserve.

**Comment by javiercn** [comment]
This is a functionality of the debugger, if it doesn't work, we shouldn't workaround anything in the framework. We should report it to the right folks to make sure that the experience is fixed. The instructions below are for VS, but applies to VS Code and VS 4 Mac in the same way.

https://docs.microsoft.com/en-us/visualstudio/debugger/managing-exceptions-with-the-debugger?view=vs-2022#tell-the-debugger-to-break-when-an-exception-is-thrown

**Comment by jasonswearingen** [comment]
@javiercn related, do you know if unhandled exceptions are able to be breakpoint triggered in MAUI Blazor WebView?   The description of this issue  https://github.com/dotnet/maui/issues/4441 imples that it works, but upon closing it is not clear that this is actually working
I ask because I might consider attempting BlazorWebview as a dev workflow if thrown exceptions get too frustrating in the short term

**Comment by javiercn** [comment]
@jasonswearingen as far as I know works for both exceptions thrown from JS (if you have the dev tools open and configured to break on thrown JS exceptions) and exceptions thrown from .NET if you have configured the .NET debugger to break on them.

**Comment by TanayParikh** [comment]
> This is a functionality of the debugger, if it doesn't work, we shouldn't workaround anything in the framework. We should report it to the right folks to make sure that the experience is fixed.

I've reported it on the VS side here: https://developercommunity.visualstudio.com/t/Blazor-WASM-Debugging-Doesnt-Catch-Unha/10079630?port=1025&fsid=33a7df2e-9e57-4282-b108-21743a1e9a1a&entry=problem
